tests/api2/test_audit_basic.py

```diff
@@ -324,47 +315,58 @@ def test_audit_timestamps(self, svc):
 
 @pytest.mark.skipif(not ha, reason="Skip HA tests")
 class TestAuditOpsHA:
-    def test_audit_ha_query(self, standby_user):
-        name = standby_user
-        remote_user = call(
-            'failover.call_remote', 'user.query',
-            [[["username", "=", name]]],
-            {'raise_connect_error': False, 'timeout': 2, 'connect_timeout': 2}
-        )
-        assert remote_user != []
-
-        # Handle delays in the audit database
-        remote_audit_entry = []
-        tries = 3
-        while tries > 0 and remote_audit_entry == []:
-            sleep(1)
-            remote_audit_entry = call('audit.query', {
-                "query-filters": [["event_data.description", "$", name]],
-                "query-options": {"select": ["event_data", "success"]},
-                "remote_controller": True
-            })
-            if remote_audit_entry != []:
-                break
-            tries -= 1
-
-        assert tries > 0, "Failed to get expected audit entry"
-        assert remote_audit_entry != []
-        params = remote_audit_entry[0]['event_data']['params'][0]
-        assert params['username'] == name
-
-    def test_audit_ha_export(self, standby_user):
+    @pytest.mark.parametrize('remote_available', [True, False])
+    def test_audit_ha_query(self, standby_audit_event, remote_available):
+        assert standby_audit_event, "Missing event data"
```

Comment:
A fixture that fails to set up will fail all tests that depend on it

Reply:
Refactoring.